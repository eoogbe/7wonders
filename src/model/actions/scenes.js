/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const VIEW_WONDER_CHOOSER = 'VIEW_WONDER_CHOOSER';

export const VIEW_WONDER_PANEL = 'VIEW_WONDER_PANEL';

export const VIEW_GAME_BOARD = 'VIEW_GAME_BOARD';

export const START_AGE = 'START_AGE';

export const SCORE_GAME = 'SCORE_GAME';

export const QUIT = 'QUIT';

export const viewWonderChooser = (playerId) => ({
  type: VIEW_WONDER_CHOOSER,
  payload: {
    playerId,
  },
});

export const viewWonderPanel = () => ({
  type: VIEW_WONDER_PANEL,
});

export const viewGameBoard = () => ({
  type: VIEW_GAME_BOARD,
});

export const startAge = () => ({
  type: START_AGE,
});

export const scoreGame = () => ({
  type: SCORE_GAME,
});

export const quit = () => ({
  type: QUIT,
});
