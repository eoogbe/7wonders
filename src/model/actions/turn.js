/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const BUILD_STRUCTURE = 'BUILD_STRUCTURE';

export const BUILD_WONDER = 'BUILD_WONDER';

export const ADD_MONEY = 'ADD_MONEY';

export const SELECT_HAND_CARD = 'SELECT_HAND_CARD';

export const CHOOSE_NEIGHBOR = 'CHOOSE_NEIGHBOR';

export const COPY_GUILD = 'COPY_GUILD';

export const buildStructure = (cardId) => ({
  type: BUILD_STRUCTURE,
  payload: {
    cardId,
  },
});

export const buildFreeStructure = () => ({
  type: BUILD_STRUCTURE,
  payload: {
    isFree: true,
  },
});

export const buildWonder = () => ({
  type: BUILD_WONDER,
});

export const addMoney = () => ({
  type: ADD_MONEY,
});

export const selectHandCard = (cardId, cardIdx) => ({
  type: SELECT_HAND_CARD,
  payload: {
    cardId,
    cardIdx,
  },
});

export const chooseNeighbor = (costType) => ({
  type: CHOOSE_NEIGHBOR,
  payload: {
    costType,
  },
});

export const copyGuild = (cardId) => ({
  type: COPY_GUILD,
  payload: {
    cardId,
  },
});
