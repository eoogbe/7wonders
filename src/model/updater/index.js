/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { concat, decorate, withDefaultState } from 'redux-fp';
import ages from './ages';
import app, { initialAppState } from './app';
import copyAiGuild from './copyAiGuild';
import hands from './hands';
import military from './military';
import neighbors from './neighbors';
import players from './players';
import takeAiTurns from './takeAiTurns';
import trade from './trade';
import wonder from './wonder';

const initialState = {
  ages: [],
  players: {},
  app: initialAppState,
};

export default decorate(
  withDefaultState(initialState),
  concat(
    wonder,
    neighbors,
    takeAiTurns,
    ages,
    hands,
    players,
    military,
    copyAiGuild,
    app,
    trade
  )
);
