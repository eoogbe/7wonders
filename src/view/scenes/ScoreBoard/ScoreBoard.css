/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

.ScoreBoard {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.ScoreBoard-scoreTable {
  font-size: 1rem;
  margin-bottom: 0.5em;
  border-collapse: collapse;
}

.ScoreBoard-caption {
  font-size: 1.25rem;
  margin-bottom: 0.25em;
}

.ScoreBoard-playerHeader {
  padding: 1px 3px;
  border-color: #BDBDBD;
  border-style: solid;
  border-width: 0 2px 2px 0;
  text-align: left;
  font-weight: normal;
}

.ScoreBoard-colHeader {
  padding: 1px 3px;
  border-color: #BDBDBD;
  border-style: solid;
  border-width: 0 0 2px 0;
  text-align: center;
  font-weight: normal;
}

.ScoreBoard-totalHeader {
  padding: 1px 3px;
  border-color: #BDBDBD;
  border-style: solid;
  border-width: 0 0 2px 2px;
  text-align: center;
  font-weight: normal;
}

.ScoreBoard-rowHeader {
  padding: 1px 2px;
  border-color: #BDBDBD;
  border-style: solid;
  border-width: 0 2px 1px 0;
  text-align: left;
  font-weight: normal;
}

.ScoreBoard-row:first-child th,
.ScoreBoard-row:first-child td {
  font-weight: bold;
}

.ScoreBoard-row:last-child th,
.ScoreBoard-row:last-child td {
  border-bottom-width: 0;
}

.ScoreBoard-cell {
  padding: 1px 2px;
  text-align: center;
  border-bottom: 1px solid #BDBDBD;
}

.ScoreBoard-totalCell {
  padding: 1px 2px;
  border-color: #BDBDBD;
  border-style: solid;
  border-width: 0 0 1px 1px;
  text-align: center;
}

.ScoreBoard-buttonPanel {
  display: flex;
  justify-content: space-around;
  width: 50%;
}
