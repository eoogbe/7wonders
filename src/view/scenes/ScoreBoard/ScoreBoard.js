/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { compose } from 'ramda';
import { useDispatch, useSelector } from 'react-redux';
import * as actions from 'model/actions';
import { getPlayerScores } from 'presentation/players';
import QuitButton from 'view/components/QuitButton';
import * as theme from 'view/theme';
import './ScoreBoard.css';

export const CELL_COLORS = {
  MILITARY: theme.lightRed,
  MONEY: theme.lightYellow,
  WONDER: theme.lightBrown,
  CIVIC: theme.lightBlue,
  SCIENCE: theme.lightGreen,
  TRADE: theme.lightYellow,
  GUILD: theme.lightPurple,
  TOTAL: 'rgb(255, 255, 255)',
};

const getScoreStyle = (type) => ({
  backgroundColor: CELL_COLORS[type],
});

const renderScore = ({ score, type }, idx) => {
  const className =
    type === 'TOTAL' ? 'ScoreBoard-totalCell' : 'ScoreBoard-cell';
  return (
    <td
      key={idx}
      className={className}
      style={getScoreStyle(type)}
    >
      {score}
    </td>
  );
};

const renderPlayerRow = ({ player, scores }) => (
  <tr key={player} className="ScoreBoard-row">
    <th className="ScoreBoard-rowHeader" scope="row">{player}</th>
    {scores.map(renderScore)}
  </tr>
);

const ScoreBoard = () => {
  const { winner, playerScores } = useSelector(getPlayerScores);

  const dispatch = useDispatch();
  const viewGameBoard = compose(dispatch, actions.viewGameBoard);

  const buttonStyle = {
    fontSize: '1.25em',
    width: '8em',
    backgroundColor: theme.primaryColor,
    color: '#FFF',
  };

  return (
    <div className="ScoreBoard">
      <table className="ScoreBoard-scoreTable">
        <caption className="ScoreBoard-caption">
          <b className="keyword">Winner:</b> {winner}
        </caption>
        <thead>
          <tr>
            <th className="ScoreBoard-playerHeader" scope="col">Player</th>
            <th
              className="ScoreBoard-colHeader"
              style={getScoreStyle('MILITARY')}
              scope="col"
            >
              Military
            </th>
            <th
              className="ScoreBoard-colHeader"
              style={getScoreStyle('MONEY')}
              scope="col"
            >
              Money
            </th>
            <th
              className="ScoreBoard-colHeader"
              style={getScoreStyle('WONDER')}
              scope="col"
            >
              Wonder
            </th>
            <th
              className="ScoreBoard-colHeader"
              style={getScoreStyle('CIVIC')}
              scope="col"
            >
              Civic
            </th>
            <th
              className="ScoreBoard-colHeader"
              style={getScoreStyle('SCIENCE')}
              scope="col"
            >
              Science
            </th>
            <th
              className="ScoreBoard-colHeader"
              style={getScoreStyle('TRADE')}
              scope="col"
            >
              Trade
            </th>
            <th
              className="ScoreBoard-colHeader"
              style={getScoreStyle('GUILD')}
              scope="col"
            >
              Guild
            </th>
            <th className="ScoreBoard-totalHeader" scope="col">Total</th>
          </tr>
        </thead>
        <tbody>
          {playerScores.map(renderPlayerRow)}
        </tbody>
      </table>
      <div className="ScoreBoard-buttonPanel">
        <QuitButton style={buttonStyle} />
        <button
          className="primary-button"
          style={buttonStyle}
          type="button"
          onClick={viewGameBoard}
        >
          Return to game
        </button>
      </div>
    </div>
  );
};

export default ScoreBoard;
