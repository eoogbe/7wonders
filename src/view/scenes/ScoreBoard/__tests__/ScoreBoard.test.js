/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import PLAYERS from '__fixtures__/players';
import { viewGameBoard } from 'model/actions';
import ScoreBoard from '../ScoreBoard';

const mockStore = configureStore();

it('renders without crashing', () => {
  const store = mockStore(createState());
  const tree =
    renderer.create(
      <Provider store={store}>
        <ScoreBoard />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('views the game board', () => {
  const store = mockStore(createState());
  const tree = mount(
    <Provider store={store}>
      <ScoreBoard />
    </Provider>
  );
  tree.find('button').last().simulate('click');
  expect(store.getActions()).toEqual([viewGameBoard()]);
});

const createState = () => ({
  app: { currentPlayer: 'PLAYER1' },
  players: PLAYERS,
})
