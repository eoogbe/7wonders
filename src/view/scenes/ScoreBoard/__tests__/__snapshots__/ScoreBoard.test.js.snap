// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="ScoreBoard"
>
  <table
    className="ScoreBoard-scoreTable"
  >
    <caption
      className="ScoreBoard-caption"
    >
      <b
        className="keyword"
      >
        Winner:
      </b>
       
      TEMPLE_OF_ARTEMIS_IN_EPHESUS
    </caption>
    <thead>
      <tr>
        <th
          className="ScoreBoard-playerHeader"
          scope="col"
        >
          Player
        </th>
        <th
          className="ScoreBoard-colHeader"
          scope="col"
          style={
            Object {
              "backgroundColor": "rgb(239,83,80, 0.7)",
            }
          }
        >
          Military
        </th>
        <th
          className="ScoreBoard-colHeader"
          scope="col"
          style={
            Object {
              "backgroundColor": "rgb(255,235,59, 0.7)",
            }
          }
        >
          Money
        </th>
        <th
          className="ScoreBoard-colHeader"
          scope="col"
          style={
            Object {
              "backgroundColor": "rgb(141,110,99, 0.7)",
            }
          }
        >
          Wonder
        </th>
        <th
          className="ScoreBoard-colHeader"
          scope="col"
          style={
            Object {
              "backgroundColor": "rgb(33,150,243, 0.7)",
            }
          }
        >
          Civic
        </th>
        <th
          className="ScoreBoard-colHeader"
          scope="col"
          style={
            Object {
              "backgroundColor": "rgb(76,175,80, 0.7)",
            }
          }
        >
          Science
        </th>
        <th
          className="ScoreBoard-colHeader"
          scope="col"
          style={
            Object {
              "backgroundColor": "rgb(255,235,59, 0.7)",
            }
          }
        >
          Trade
        </th>
        <th
          className="ScoreBoard-colHeader"
          scope="col"
          style={
            Object {
              "backgroundColor": "rgb(171,71,188, 0.7)",
            }
          }
        >
          Guild
        </th>
        <th
          className="ScoreBoard-totalHeader"
          scope="col"
        >
          Total
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="ScoreBoard-row"
      >
        <th
          className="ScoreBoard-rowHeader"
          scope="row"
        >
          PYRAMIDS_OF_GIZA
        </th>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(239,83,80, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(255,235,59, 0.7)",
            }
          }
        >
          1
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(141,110,99, 0.7)",
            }
          }
        >
          3
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(33,150,243, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(76,175,80, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(255,235,59, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(171,71,188, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-totalCell"
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
            }
          }
        >
          4
        </td>
      </tr>
      <tr
        className="ScoreBoard-row"
      >
        <th
          className="ScoreBoard-rowHeader"
          scope="row"
        >
          COLOSSUS_OF_RHODES
        </th>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(239,83,80, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(255,235,59, 0.7)",
            }
          }
        >
          1
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(141,110,99, 0.7)",
            }
          }
        >
          3
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(33,150,243, 0.7)",
            }
          }
        >
          5
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(76,175,80, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(255,235,59, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(171,71,188, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-totalCell"
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
            }
          }
        >
          9
        </td>
      </tr>
      <tr
        className="ScoreBoard-row"
      >
        <th
          className="ScoreBoard-rowHeader"
          scope="row"
        >
          TEMPLE_OF_ARTEMIS_IN_EPHESUS
        </th>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(239,83,80, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(255,235,59, 0.7)",
            }
          }
        >
          1
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(141,110,99, 0.7)",
            }
          }
        >
          10
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(33,150,243, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(76,175,80, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(255,235,59, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-cell"
          style={
            Object {
              "backgroundColor": "rgb(171,71,188, 0.7)",
            }
          }
        >
          0
        </td>
        <td
          className="ScoreBoard-totalCell"
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
            }
          }
        >
          11
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="ScoreBoard-buttonPanel"
  >
    <button
      className="QuitButton"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(93, 64, 55)",
          "color": "#FFF",
          "fontSize": "1.25em",
          "width": "8em",
        }
      }
      type="button"
    >
      Quit
    </button>
    <button
      className="primary-button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(93, 64, 55)",
          "color": "#FFF",
          "fontSize": "1.25em",
          "width": "8em",
        }
      }
      type="button"
    >
      Return to game
    </button>
  </div>
</div>
`;
